src/playerState/index.js

```diff
@@ -0,0 +1,79 @@
+import React, {
+  useReducer,
+  createContext,
+  useContext,
+  useCallback,
+} from "react";
+
+import { reducer, PlayerActions } from "./reducer";
+
+const initialContext = {
+  name: "",
+  hp: 0,
+  maxhp: 0,
+  attack: 0,
+  defense: 0,
+};
+
+const StateContext = createContext(initialContext);
+const DispatchContext = createContext(undefined);
+
+export const PlayerContextProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, initialContext);
```

Comment:
Ill have to look a bit more into the useReducer function. But love how this is set up! Great job.